gobblin-core/src/main/java/gobblin/writer/AvroHdfsTimePartitionedWriter.java

```diff
@@ -172,9 +170,11 @@ public void write(GenericRecord record) throws IOException {
     // If the path is in pathToWriterMap simply retrieve the writer, and write the record
     if (!this.pathToWriterMap.containsKey(writerOutputPath)) {
 
-      LOG.info("Creating a new DataWriter for path: " + new Path(this.baseFilePath, writerOutputPath));
+      Path writerOutputPathWithBase = new Path(this.baseFilePath, writerOutputPath);
+
+      LOG.info("Creating a new DataWriter for path: " + writerOutputPathWithBase);
```

Comment:
I see. Then I think you can just have writerOutputPath contain baseFilePath. There doesn't seem to be any inconsistency since `WriterUtils.getWriterOutputDir` won't append baseFilePath to the end of the dir if property `writer.file.path` is already set, which is the case when creating the DataWriter. So you can simply remove the redundant `baseFilePath` from `LOG.info`.

Also, the name `baseFilePath` is very confusing, please rename it to `datasetName`.